opcua/ua/uatypes.py

```diff
@@ -163,12 +163,35 @@ class EventNotifier(_MaskEnum):
 
 class Guid(FrozenClass):
 
-    def __init__(self):
-        self.uuid = uuid.uuid4()
+    def __init__(self, value=None):
+        if value is None or value == "":
+            # if no string is supplied, create a new UUID
+            self.uuid = uuid.uuid4()
+        else:
+            try:
+                # attempt to make a UUID with the supplied value, if hex string has errors use a default string
+                self.uuid = uuid.UUID(value)
+            except ValueError as e:
+                # TODO should notify user of error, but this module has no logger, raise UaError stops execution
+                self.uuid = uuid.UUID("00000000-0000-0000-0000-000000000000")
```

Comment:
If uuid module receives a bad value it raises ValueError. I did it this way so that the program doesn't crash if the Guid is invalid. The main reason for this is for apps like modeler. If you make a Guid in modeler and type something wrong you will now just get a default value instead of crashing the entire program.

If it's better to crash the program than set a default then I can change it.